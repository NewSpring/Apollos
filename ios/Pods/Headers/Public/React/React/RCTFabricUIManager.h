../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h