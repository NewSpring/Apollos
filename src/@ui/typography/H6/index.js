import { Platform, Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import styled from '@ui/styled';
import { withPlaceholder, Typography } from '@ui/Placeholder';

const styles = styled(
  ({ theme }) => ({
    fontSize: theme.helpers.rem(0.875),
    fontFamily: theme.typography.fontFamilySans.bold.default,
    color: theme.colors.text.primary,
    ...Platform.select({
      ios: {
        paddingTop: theme.helpers.rem(0.25),
        paddingBottom: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(0.875, 1.014),
      },
      android: {
        paddingTop: theme.helpers.rem(0.125),
        lineHeight: theme.helpers.verticalRhythm(0.875, 1.214),
      },
      web: {
        paddingTop: theme.helpers.rem(0.3),
        paddingBottom: theme.helpers.rem(0.075),
        lineHeight: theme.helpers.verticalRhythm(0.875, 1.05),
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
      },
    }),
  }),
  'H6',
);

const H6 = compose(
  setPropTypes({
    isLoading: PropTypes.bool, // display loading placeholder
    ...Text.propTypes,
  }),
  styles,
  withPlaceholder(Typography, { width: '50%' }),
  pure,
)(Text);

export default H6;
