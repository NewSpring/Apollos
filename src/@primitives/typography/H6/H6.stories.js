import React from 'react';

import { storiesOf } from '@storybook/react-native';

import ThemeProvider from '@primitives/ThemeProvider';
import H6 from './';

storiesOf('Typography/Headings', module)
  .add('H6', () => (
    <ThemeProvider>
      <H6>{'"Radical obedience to Christ is not easy... It\'s not comfort, not health, not wealth, and not prosperity in this world. Radical obedience to Christ risks losing all these things. But in the end, such risk finds its reward in Christ. And he is more than enough for us." â€• David Platt'}</H6>
    </ThemeProvider>
  ));
