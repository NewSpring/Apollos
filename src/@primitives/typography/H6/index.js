import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';
import rem from '@utils/remUnit';
import verticalRhythm from '@utils/verticalRhythm';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledH6 = styled(({ theme }) => ({
  color: theme.baseFontColor,
  fontSize: rem(0.875),
  lineHeight: verticalRhythm(0.875, 1.02),
}))(Text);

const H6 = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledH6
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledH6>
));

export default H6;
