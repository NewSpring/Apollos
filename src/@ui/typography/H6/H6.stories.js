import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';


import H6 from './';

storiesOf('typography/H6', module)
  .add('Default', () => (
    <H6>{'"Radical obedience to Christ is not easy... It\'s not comfort, not health, not wealth, and not prosperity in this world. Radical obedience to Christ risks losing all these things. But in the end, such risk finds its reward in Christ. And He is more than enough for us." ― David Platt'}</H6>
  ))
  .add('Border Box – platform testing', () => {
    const border = { borderWidth: 1, borderStyle: 'solid' };

    return (
      <View>
        <H6 style={border}>Heading 6</H6>
        <H6 style={border}>{'"Radical obedience to Christ is not easy... It\'s not comfort, not health, not wealth, and not prosperity in this world. Radical obedience to Christ risks losing all these things. But in the end, such risk finds its reward in Christ. And He is more than enough for us." ― David Platt'}</H6>
      </View>
    );
  });
