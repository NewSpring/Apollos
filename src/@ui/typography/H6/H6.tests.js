import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import H6 from './';

describe('the H6 component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Providers>
        <H6>Default H6 text</H6>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H6 style={salmon}>Salmon text</H6>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <H6 accessible={false}>{'"Radical obedience to Christ is not easy... It\'s not comfort, not health, not wealth, and not prosperity in this world. Radical obedience to Christ risks losing all these things. But in the end, such risk finds its reward in Christ. And he is more than enough for us." â€• David Platt'}</H6>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
