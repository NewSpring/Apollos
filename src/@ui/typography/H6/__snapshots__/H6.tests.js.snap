// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H6 component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Bold",
      "fontSize": 15.75,
      "lineHeight": 17.75,
      "paddingBottom": 1.8,
      "paddingTop": 4.5,
    }
  }
>
  "Radical obedience to Christ is not easy... It's not comfort, not health, not wealth, and not prosperity in this world. Radical obedience to Christ risks losing all these things. But in the end, such risk finds its reward in Christ. And he is more than enough for us." â€• David Platt
</Text>
`;

exports[`the H6 component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "salmon",
      "fontFamily": "Colfax-Bold",
      "fontSize": 15.75,
      "lineHeight": 17.75,
      "paddingBottom": 1.8,
      "paddingTop": 4.5,
    }
  }
>
  Salmon text
</Text>
`;

exports[`the H6 component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Bold",
      "fontSize": 15.75,
      "lineHeight": 17.75,
      "paddingBottom": 1.8,
      "paddingTop": 4.5,
    }
  }
>
  Default H6 text
</Text>
`;
